src/Api/ErrorParser/JsonParserTrait.php

```diff
@@ -15,10 +15,14 @@ trait JsonParserTrait
     private function genericHandler(ResponseInterface $response)
     {
         $code = (string) $response->getStatusCode();
+        if ($queryError = $response->getHeaderLine('x-amzn-query-error')) {
+            $parts = explode(';', $queryError);
+            $error = (isset($parts) and !empty(trim($parts[0]))) ? $parts[0] : null;
```

Comment:
We need to add something for type here, where we parse the query error and return type from $parts[1].  That should be passed into 'type' on line 23/27 below.

Reply:
will do that.